File: backend/app/api/seed.py
import uuid
from datetime import datetime
from flask import Blueprint, jsonify
from backend.app.db.connection import query_one, execute
from backend.app.api.auth import hash_password
from backend.app.db.seed_data import DEMO_REPO_NAME, DEMO_REPO_URL
from backend.app.analysis.service import analyze_repository_full

seed_bp = Blueprint("seed", __name__)

@seed_bp.route("/api/seed", methods=["POST"])
def seed_database():
    user = query_one("SELECT * FROM users WHERE email = 'demo@regit.dev'")
    if not user:
        user_id = str(uuid.uuid4())
        pw = hash_password("regit-demo-123")
        execute(
            """
            INSERT INTO users (id, email, name, password_hash, avatar_hue, created_at)
            VALUES (%s, 'demo@regit.dev', 'Rishab Dixit', %s, 95, %s)
            """,
            (user_id, pw, datetime.now())
        )
        user = {"id": user_id}
    else:
        user_id = user["id"]

    demo_repo = query_one(
        "SELECT * FROM repositories WHERE user_id = %s AND is_demo = true",
        (user_id,)
    )
    if not demo_repo:
        repo_id = str(uuid.uuid4())
        now = datetime.now()
        execute(
            """
            INSERT INTO repositories (id, user_id, name, github_url, branch, description, status, is_demo, created_at, updated_at)
            VALUES (%s, %s, %s, %s, 'main', 'Intentionally-designed demo: rename fallout, contract breaks, missing auth, cleanup targets.', 'pending', true, %s, %s)
            """,
            (repo_id, user_id, DEMO_REPO_NAME, DEMO_REPO_URL, now, now)
        )
        demo_repo_id = repo_id
    else:
        demo_repo_id = demo_repo["id"]

    # Run analysis
    analyze_repository_full(demo_repo_id)

    return jsonify({
        "ok": True,
        "userId": user_id,
        "demoRepoId": demo_repo_id
    })

File: src/lib/seed.ts
import { db } from "@/db";
import { repositories, users } from "@/db/schema";
import { eq } from "drizzle-orm";
import { hashPassword } from "./auth";
import { DEMO_REPO_NAME, DEMO_REPO_URL } from "./demo-repo";
import { analyzeRepository } from "./service";

export async function ensureSeed(): Promise<{ userId: string; demoRepoId: string }> {
  let userRows = await db.select().from(users).where(eq(users.email, "demo@regit.dev")).limit(1);
  let userId: string;
  if (userRows.length === 0) {
    const passwordHash = await hashPassword("regit-demo-123");
    const inserted = await db
      .insert(users)
      .values({ email: "demo@regit.dev", name: "Rishab Dixit", passwordHash, avatarHue: 95 })
      .returning({ id: users.id });
    userId = inserted[0].id;
  } else {
    userId = userRows[0].id;
  }

  const existingRepos = await db
    .select()
    .from(repositories)
    .where(eq(repositories.userId, userId))
    .limit(10);

  let demo = existingRepos.find((r) => r.isDemo);
  if (!demo) {
    const inserted = await db
      .insert(repositories)
      .values({
        userId,
        name: DEMO_REPO_NAME,
        githubUrl: DEMO_REPO_URL,
        branch: "main",
        description: "Intentionally-designed demo: rename fallout, contract breaks, missing auth, cleanup targets.",
        status: "pending",
        isDemo: true,
      })
      .returning();
    demo = inserted[0];
  }

  if (demo.status !== "ready") {
    try {
      await analyzeRepository(demo.id);
    } catch (e) {
      console.error("seed analysis failed", e);
    }
  }

  // second sample repo for liveliness
  if (existingRepos.length < 2) {
    const second = await db
      .insert(repositories)
      .values({
        userId,
        name: "billing-api",
        githubUrl: "https://github.com/rishabdixit4021/billing-api",
        branch: "main",
        description: "Sample service connected to preview multi-repo impact. Auto-analyzed scaffold.",
        status: "pending",
        isDemo: false,
      })
      .returning();
    try {
      await analyzeRepository(second[0].id);
    } catch (e) {
      console.error("second seed failed", e);
    }
  }

  return { userId, demoRepoId: demo.id };
}
